__tests__/app.test.js

```diff
@@ -416,3 +418,93 @@ describe("GET /api/users", () => {
+                })
+            })
+            expect(body.articles).toBeSortedBy('created_at', {descending: true})
+        })
+    })
+    test("responds with articles filtered by given topic query", () => {
+        return request(app).get('/api/articles/?topic=mitch')
+        .expect(200)
+        .then(({ body }) => {
+            expect(body.articles.length).toBe(12)
+            body.articles.forEach((article) => {
+                expect(article).toMatchObject({
+                    author: expect.any(String),
+                    title: expect.any(String),
+                    article_id: expect.any(Number),
+                    topic: 'mitch',
+                    created_at: expect.any(String),
+                    votes: expect.any(Number),
+                    article_img_url: expect.any(String),
+                    comment_count: expect.any(Number)
+                })
+            })
+            expect(body.articles).toBeSortedBy('created_at', {descending: true})
+        })
+    })
+    test("ignores invalid query keys", () => {
+        return request(app).get('/api/articles/?topic=mitch&hungry=true')
+        .expect(200)
+        .then(({ body }) => {
+            expect(body.articles.length).toBe(12)
+            body.articles.forEach((article) => {
+                expect(article).toMatchObject({
+                    author: expect.any(String),
+                    title: expect.any(String),
+                    article_id: expect.any(Number),
+                    topic: 'mitch',
+                    created_at: expect.any(String),
+                    votes: expect.any(Number),
+                    article_img_url: expect.any(String),
+                    comment_count: expect.any(Number)
+                })
+            })
+            expect(body.articles).toBeSortedBy('created_at', {descending: true})
+        })
+    })
+    test("responds with all articles if given no query key", () => {
+        return request(app).get('/api/articles')
+        .expect(200)
+        .then(({ body }) => {
+            expect(body.articles.length).toBe(13)
+            body.articles.forEach((article) => {
+                expect(article).toMatchObject({
+                    author: expect.any(String),
+                    title: expect.any(String),
+                    article_id: expect.any(Number),
+                    topic: expect.any(String),
+                    created_at: expect.any(String),
+                    votes: expect.any(Number),
+                    article_img_url: expect.any(String),
+                    comment_count: expect.any(Number)
+                })
+            })
+        })
+    })
+
+    test("responds with 404 if topic doesnt exist", () => {
+        return request(app).get('/api/articles/?topic=dogs')
+        .expect(404)
+        .then(({ body }) => {
+            expect(body.msg).toBe(`topic with value dogs does not exist in articles`)
+        })
+    })
+
```

Comment:
We may also want to consider what will happen with a valid query that exists but happens to have no articles associated with it. 200 empty response.